wire-library/wire-runtime-swift/src/main/swift/ProtoReader.swift

```diff
@@ -103,6 +103,13 @@ public final class ProtoReader {
     }
 
     deinit {
+         //If decoding fails for any reason, there may be unknownFields to
+         //clean up that were not caught by the defer block in endMessage
+        if messageStackIndex > -1 {
```

Comment:
I went down that route but most of the time `messageStackIndex` is `-1`. That would cause the range to throw a `Can't form Range with upperBound < lowerBound` exception.

Reply:
Ah, good to know. LGTM then.